mmdet/models/necks/yolox_asff_pafpn.py

```diff
@@ -0,0 +1,303 @@
+        elif self.level == 2:
+            level_0_compressed = self.compress_level_0(x_level_0)
+            level_0_resized = F.interpolate(
+                level_0_compressed, scale_factor=4, mode='nearest')
+            x_level_1_compressed = self.compress_level_1(x_level_1)
+            level_1_resized = F.interpolate(
+                x_level_1_compressed, scale_factor=2, mode='nearest')
+            level_2_resized = x_level_2
+
+        level_0_weight_v = self.weight_level_0(level_0_resized)
+        level_1_weight_v = self.weight_level_1(level_1_resized)
+        level_2_weight_v = self.weight_level_2(level_2_resized)
+
+        levels_weight_v = torch.cat(
+            (level_0_weight_v, level_1_weight_v, level_2_weight_v), 1)
+        levels_weight = self.weight_levels(levels_weight_v)
+        levels_weight = F.softmax(levels_weight, dim=1)
+
+        fused_out_reduced = (level_0_resized * levels_weight[:, 0:1, :, :]) + (
+            level_1_resized * levels_weight[:, 1:2, :, :]) + (
+                level_2_resized * levels_weight[:, 2:, :, :])
+        out = self.expand(fused_out_reduced)
+
+        return out
+
+
+@NECKS.register_module()
+class YOLOXASFFPAFPN(YOLOXPAFPN):
+    """Path Aggregation Network used in `YOLOX-PAI.
+
+    <https://arxiv.org/abs/2208.13040>`_.
+
+    Args:
+        asff_channel (int): The hidden channel of the attention layer in
+            ASFF. Default: 2.
+        expand_kernel (int): Expand kernel size of the expand layer.
+            Default: 3.
+        norm_cfg (dict): Config dict for normalization layer.
+            Default: dict(type='BN', momentum=0.03, eps=0.001)
+        act_cfg (dict): Config dict for activation layer.
+            Default: dict(type='SiLU')
+    """
+
+    def __init__(self,
+                 *args,
+                 asff_channel=2,
+                 expand_kernel=3,
+                 norm_cfg=dict(type='BN', momentum=0.03, eps=0.001),
+                 act_cfg=dict(type='SiLU'),
+                 **kwargs):
+        super().__init__(*args, norm_cfg=norm_cfg, act_cfg=act_cfg, **kwargs)
+        assert len(self.in_channels) == 3,\
+            'len(in_channels) should be set to 3.'
+        # todo: handle len(self.in_channels) > 3
+
+        self.asff_1 = ASFF(
+            self.in_channels,
+            level=0,
+            asff_channel=asff_channel,
+            expand_kernel=expand_kernel,
+            norm_cfg=norm_cfg,
+            act_cfg=act_cfg,
+        )
+        self.asff_2 = ASFF(
+            self.in_channels,
+            level=1,
+            asff_channel=asff_channel,
+            expand_kernel=expand_kernel,
+            norm_cfg=norm_cfg,
+            act_cfg=act_cfg,
+        )
+        self.asff_3 = ASFF(
+            self.in_channels,
+            level=2,
+            asff_channel=asff_channel,
+            expand_kernel=expand_kernel,
+            norm_cfg=norm_cfg,
+            act_cfg=act_cfg,
+        )
+
+    def forward(self, inputs):
```

Comment:
`feat_heigh`, `feat_height` -> `feat_high` in `yolox_asff_pafpn.py` and `yolox_pafpn.py`